packages/ez-react/src/recipes/area/AreaChart.stories.tsx

```diff
@@ -1,34 +1,76 @@
 import React from 'react';
 import { Meta, Story } from '@storybook/react';
 import { AreaChart, AreaChartProps } from '@/recipes/area/AreaChart';
-import { baseChartArgTypes, ChartWrapper } from '../../lib/storybook-utils';
+import {
+  ChartWrapper,
+  flattenArgs,
+  unFlattenArgs,
+  baseChartArgTypes,
+  markerArgTypesOptions,
+} from '../../lib/storybook-utils';
 import { colors, evolutionData } from 'eazychart-dev/storybook/data';
 import { MultiAreaChart, MultiAreaChartProps } from './MultiAreaChart';
 
+const areaChartArgTypes = {
+  'area.stroke': {
+    control: { type: 'color' },
+    table: { category: 'Area props', defaultValue: { summary: 'color' } },
+    description: 'Sets the stroke color',
+    if: { arg: 'yAxis', truthy: false },
+  },
+  'area.strokeWidth': {
+    control: { type: 'number' },
+    table: { category: 'Area props', defaultValue: { summary: '2px' } },
+    description: 'Sets the stroke width',
+    if: { arg: 'yAxis', truthy: false },
+  },
+  'area.fill': {
+    control: { type: 'color' },
+    table: { category: 'Area props', defaultValue: { summary: '#26547c' } },
+    description: 'Sets the area color',
+    if: { arg: 'yAxis', truthy: false },
+  },
+  yAxis: {
+    control: { type: 'object' },
+    table: {
+      category: 'Multi chart Y Axis Options',
+      defaultValue: { summary: 'yValues' },
+    },
+    description: 'Sets the Y axis domain keys and title for multi chart',
+    //trick to make this argument disappear in the single (default) chart
+    if: { arg: 'yAxis', truthy: true },
+  },
+  ...markerArgTypesOptions,
+  ...baseChartArgTypes,
+};
+
 const meta: Meta = {
   id: '2',
   title: 'React/Area Chart',
   component: AreaChart,
   parameters: {
     controls: { expanded: true },
   },
-  argTypes: baseChartArgTypes,
+  argTypes: areaChartArgTypes,
 };
 
 export default meta;
 
 const Template: Story<AreaChartProps> = (args) => {
```

Comment:
Let's add a wrapper func in `storybook-utils.tsx` :
```
export buildTemplate = <T>(storyFn: Story<T>): Story<T> => {
  return (args) => {
    const extendedArgs = unFlattenArgs(args);
    return storyFn(extendedArgs);
  };
}
```
and then do the following usual call for each template : 
```
const Template: Story<AreaChartProps> = buildTemplate((args) => {
  return (
    <ChartWrapper>
      <AreaChart {...args} />
    </ChartWrapper>
  );
});
```
This will allow us to do any processing on the args, while keeping storybook stories ignorant about it.